data/STARmap_plus/STARmap_plus.py

```diff
@@ -0,0 +1,163 @@
+                        f'{tmpdir}', f'{sample}_spatial.csv')
+        
+            sample_dir = f'{out_dir}/{sample}'
+            directories.append(sample_dir)
+
+            # Write out coordinates.tsv, labels.tsv and observations.tsv
+            with open(f'{tmpdir}/{sample}_spatial.csv', 'r') as f_in, \
+                open(f'{sample_dir}/labels.tsv', 'w') as f_out_labels,\
+                open(f'{sample_dir}/coordinates.tsv', 'w') as f_out_coords, \
+                open(f'{sample_dir}/observations.tsv', 'w') as f_out_obs :
+                
+                # skip first two lines
+                headline_1 = f_in.readline()
+                headline_2 = f_in.readline()
+                
+                f_out_obs.write(headline_1.replace(',','\t').replace('NAME', ''))
+
+                clusters = []
+
+                f_out_coords.write('\tx\ty\tz\n')
+                f_out_labels.write('\tMain_molecular_cell_type\tSub_molecular_cell_type\tMain_molecular_tissue_region\
+                                \tSub_molecular_tissue_region\tMolecular_spatial_cell_type\n')
+
+                for l in f_in:
+                    data = l.strip('\n').split(",")
+                    f_out_labels.write(f'{data[0]}\t{data[4]}\t{data[5]}\t{data[6]}\t{data[7]}\t{data[8]}\n')
+                    f_out_coords.write(f'{data[0]}\t{data[1]}\t{data[2]}\t{data[3]}\n')
+                    f_out_obs.write(l.replace(',','\t'))
+                    clusters.append(data[4])
+
+            n_cluster.append(len(set(clusters)))
+
+            download_data(f'https://zenodo.org/records/8327576/files/{sample}raw_expression_pd.csv?download=1', 
+                        f'{tmpdir}', f'{sample}raw_expression_pd.csv')
+
+            # Write counts.mtx
+            df = pd.read_table(f'{tmpdir}/{sample}raw_expression_pd.csv', sep=',', index_col=0)
+            features = df.index 
+            df = df.transpose()
+            mmwrite(f'{sample_dir}/counts.mtx', scipy.sparse.csr_matrix(df))
+
+            # TODO no further information from the data for features.tsv
+            # Write out features.tsv
+            with open(f'{sample_dir}/features.tsv', 'w') as f_out_features :
+                f_out_features.write('\tgene_version\n')
+                for feature in features:
+                    f_out_features.write(f'{feature}\tNA\n')
+
+        ## Metadata files
+        download_data(f'https://zenodo.org/records/8327576/files/metadata.csv?download=1', 
+                    f'{tmpdir}', 'metadata.csv')  
+
+        tmp_samples = []
+        position = []
+        patient = []
+
+        # TODO data from mouse so no patient --> should patient be NA?
+        # TODO position I assumed is the tissue type, which is here brain. Is that correct?
+        # TODO ncluster = number of main celltype cluster, they also hat sub celltypes. Is that correct?
+        with open(f'{tmpdir}/metadata.csv', 'r') as f_in:
```

Comment:
sample.csv serves only as meta-data records. It will not be directly used in the pipeline (as of now). So don't worry too much about the specific format. The information it contains is more important. 

* Using mouse number as patient is good. 

* Using tissue type as position is good

* Yes for this ncluster can be the main cell type. We can also add an additional column specifying the sub-celltypes number. (Also depending on how many subtypes there are...if too many then doesn't make much sense)